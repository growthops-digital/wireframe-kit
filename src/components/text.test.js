import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import Text from './text.js';

test('snapshot — text lead', () => {
	const result = renderer.create(
		<Text variant="lead">Foo</Text>,
	).toJSON();

	expect(result).toMatchSnapshot();
});

test('snapshot — text lead with custom classes', () => {
	const result = renderer.create(
		<Text className="text-blue-500" variant="lead">Foo</Text>,
	).toJSON();

	expect(result).toMatchSnapshot();
});

test('snapshot — text regular (explicit)', () => {
	const result = renderer.create(
		<Text variant="regular">Foo</Text>,
	).toJSON();

	expect(result).toMatchSnapshot();
});

test('snapshot — text regular (implicit)', () => {
	const result = renderer.create(
		<Text>Foo</Text>,
	).toJSON();

	expect(result).toMatchSnapshot();
});

test('snapshot — text small', () => {
	const result = renderer.create(
		<Text variant="small">Foo</Text>,
	).toJSON();

	expect(result).toMatchSnapshot();
});

test('text is rendered', () => {
	render(
		<Text variant="regular">Foo</Text>,
	);

	const result = screen.getByText('Foo');

	expect(result).toBeDefined();
});

test('text is rendered as p', () => {
	const {container} = render(
		<Text>Foo</Text>,
	);

	const result = container.querySelector('p');

	expect(result).not.toBeNull();
});

test('text is rendered as span', () => {
	const {container} = render(
		<Text as="span">Foo</Text>,
	);

	const result = container.querySelector('span');

	expect(result).not.toBeNull();
});
