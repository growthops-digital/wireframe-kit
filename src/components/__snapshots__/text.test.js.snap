// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot — text large 1`] = `
<p
  className="paragraph-large"
>
  Foo
</p>
`;

exports[`snapshot — text large with custom classes 1`] = `
<p
  className="paragraph-large text-blue-500"
>
  Foo
</p>
`;

exports[`snapshot — text regular (explicit) 1`] = `
<p
  className="paragraph-regular"
>
  Foo
</p>
`;

exports[`snapshot — text regular (implicit) 1`] = `
<p
  className="paragraph-regular"
>
  Foo
</p>
`;

exports[`snapshot — text small 1`] = `
<p
  className="paragraph-small"
>
  Foo
</p>
`;
